addon/components/sl-menu.js

```diff
@@ -453,194 +239,386 @@ export default Ember.Component.extend({
-        function() {
-            let currentIndex = null;
+    selectDown() {
+        const selectionsLength = this.get( 'selections' ).length;
 
-            // Determine index of rootNode currently on
-            this.get( 'rootNode.menu.pages' ).forEach( ( item, index ) => {
-                if ( item.label === this.get( 'activeChild.menu.label' ) ) {
-                    currentIndex = index;
-                }
-            });
-
-            return currentIndex;
+        if ( 1 === selectionsLength ) {
+            this.selectSubMenu();
+        } else if ( selectionsLength > 1 ) {
+            this.selectNext();
+        } else {
+            this.select( 0 );
         }
-    ).volatile(),
+    },
 
     /**
-     * Additional class string to be added to the element's class attribute
+     * Select a menu item in the "left" direction
+     *
+     * At the top-level of the menu, "left" corresponds to selecting the
+     * previous sibling menu item.
+     * Inside a sub-menu, "left" corresponds to parsing back to the parent item
      *
      * @function
-     * @returns {String}
+     * @returns {undefined}
      */
-    extraClassNamesString: Ember.computed(
-        'extraClassNames',
-        function() {
-            const extraClassNames = this.get( 'extraClassNames' );
-            let extraClassNamesString = '';
-
-            if ( !Ember.isNone( extraClassNames ) ) {
-                extraClassNamesString = extraClassNames.join( ' ' );
-            }
+    selectLeft() {
+        const selectionsLength = this.get( 'selections' ).length;
 
-            return extraClassNamesString;
+        if ( 1 === selectionsLength ) {
+            this.selectPrevious();
+        } else if ( selectionsLength > 1 ) {
+            this.selectParent();
         }
-    ),
+    },
 
     /**
-     * Boolean representation of showAll property
+     * Select the next sibling in the current context
      *
      * @function
-     * @returns {Boolean}
+     * @throws {ember/Error}
+     * @returns {undefined}
      */
-    showAllBoolean: Ember.computed(
-        'showAll',
-        function() {
-            return 'true' === this.get( 'showAll' );
+    selectNext() {
+        const selections = this.get( 'selections' );
+
+        // Select the first item from `items` if nothing is currently selected
+        if ( selections.length < 1 ) {
+            this.select( 0 );
+            return;
+        }
+
+        const selection = selections.objectAt( selections.length - 1 );
+        const currentItems = Ember.get( selection, 'items' );
+
+        if ( !currentItems ) {
+            throw new Ember.Error( 'Current selection items are undefined' );
         }
-    ),
+
+        if ( currentItems.length < 2 ) {
+            warn( '`selectNext` triggered with fewer than two siblings in context' );
+            return;
+        }
+
+        const currentIndex = Ember.get( selection, 'index' );
+
+        if ( Ember.typeOf( currentIndex ) !== 'number' ) {
```

Comment:
Change order of compared values per https://github.com/softlayer/ember-style-guide/issues/18